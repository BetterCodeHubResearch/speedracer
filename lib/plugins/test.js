const path = require('path')

const { createDir } = require('../.internal/util')

export default (options) => (reports, file) => {
  reports.forEach(report => {
    const dirname = path.join(path.dirname(file), options.snapshotsDirname)
    const filename = path.join(dirname, `${report.metadata.id}.speedracer`)

    return loadReport(filename).then(
      oldReport => {
        process.exit(0)
      },
      () => createDir(dirname).then(
        () => writeFile(filename, race.report.serialize())
      )
    )
  })
}
