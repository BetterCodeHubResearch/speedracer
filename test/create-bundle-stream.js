// Packages
import getStream from 'get-stream'
import test from 'ava'

// Ours
import createBundleStream from '../lib/create-bundle-stream'

test('create a bundle', async t => {
  const content = await getStream(createBundleStream('./test/fixtures/bundle.js'))
  t.snapshot(content)
})

test('create a bundle that import runtime', async t => {
  const content = await getStream(createBundleStream('./test/fixtures/bundle-runtime.js'))
  t.snapshot(content)
})

test('create a bundle that uses cjs', async t => {
  const content = await getStream(createBundleStream('./test/fixtures/bundle-cjs.js'))
  t.snapshot(content)
})
