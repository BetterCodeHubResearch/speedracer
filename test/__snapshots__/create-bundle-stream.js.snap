// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create a bundle 1`] = `
"(function () {
'use strict';

var dep = () => {
  console.log(\\"Hello Speed, I'm Racer X!\\");
};

dep();

}());
"
`;

exports[`create a bundle that import runtime 1`] = `
"(function () {
'use strict';

// eslint-disable-next-line
var speedracer = new class __SpeedRacer__ {
  end() {
    fetch('http://localhost:3001');
  }
};

speedracer.end();

}());
"
`;

exports[`create a bundle that uses cjs 1`] = `
"var run = (function () {
'use strict';

var dep = () => {
  console.log(\\"Hello Speed, I'm Racer X!\\");
};


var dep$1 = Object.freeze({
	default: dep
});

var dep$2 = ( dep$1 && dep ) || dep$1;

dep$2();

var bundleCjs = {

};

return bundleCjs;

}());
"
`;
